File: validation_tests/llvm/chill/hh/src/matmul.c
void matmul(  int M, int K, int N, double out[M][N], double A[M][K], double B[K][N] ){
  int i, j, k;
  for( i = 0 ; i < M ; i++ ){ 
    for( j = 0 ; j < N ; j++ ){
      //            out[ i * N + j ] = 0.0;
      for( k = 0 ; k < K ; k++ ){
	out[ i ][ j ] += A[ i ][ k ] * B[ k ][ j ];
      }
    }
  }
}

File: validation_tests/llvm/chill/hh/matmul.py
import chill

chill.source( "src/matmul.c" )
chill.destination( "src/matmul_modified.c" )

chill.procedure( "matmul" )
chill.loop( 0 )

#chill.known( "M >= 0" )
#chill.known( "K >= 0" )
#chill.known( "N >= 0" )

#chill.permute( [1, 3, 2] )
#chill.tile( 0, 2, 16 )

